perch/model.py

```diff
@@ -150,39 +168,90 @@ def get(cls, resource_id):
         if resource.get('type') != cls.resource_type:
             raise exceptions.NotFound()
 
+        if not include_deactivated and resource.get('state') == State.deactivated.name:
+            raise exceptions.NotFound()
+
         raise Return(cls(**resource))
 
     @classmethod
     @coroutine
-    def all(cls):
+    def all(cls, include_deactivated=False):
         """
         Get all resources
+        :param include_deactivated: Include deactivated resources in response
 
         :returns: list of Document instances
         :raises: SocketError, CouchException
         """
-        resources = yield cls.view.get(include_docs=True)
+        if include_deactivated:
+            resources = yield cls.view.get(include_docs=True)
+        else:
+            resources = yield cls.active_view.get(include_docs=True)
```

Comment:
We could have more complex keys and use `startkey` and `endkey` instead of `key`